### PR TITLE
New results api, take 2

### PR DESCRIPTION
**Pull request by boggle**

### CODE REVIEW

File: driver/src/main/java/org/neo4j/driver/v1/Entry.java [hunk 1]
```diff
@@ -21,11 +21,12 @@
...
  * Immutable pair of a key and a value
  *
- * @see PropertyMapAccessor
+ * @see MapAccessor
+ * @see RecordAccessor
  * @param <V> the Java type of the contained value
  */
 @Immutable
-public interface Property<V>
+public interface Entry<V>
```
Comment (pontusmelke) [inline driver/src/main/java/org/neo4j/driver/v1/Entry.java]: Doesn't the name clash with `java.util.Map.Entry` make it cumbersome to use?

Comment (boggle) [inline driver/src/main/java/org/neo4j/driver/v1/Entry.java]: Maybe.. I usually don't import it and just use Map.Entry directly in my Java code; that was the thinking here but I realize now that is not a good thing to assume by default. What would be a good generic alternative? We could just have a type `Pair<K, V>`? Wdyt?

Comment (boggle) [inline driver/src/main/java/org/neo4j/driver/v1/Entry.java]: Wikipedia agrees with that idea: https://en.wikipedia.org/wiki/Associative_array

**Comment by boggle** [comment]
Something is weird with the PR build, it looks like there is a hanging db server. Let's see if this clears up tomorrow.

File: driver/src/main/java/org/neo4j/driver/v1/Entry.java (hunk 1 above)
Comment (pontusmelke) [inline driver/src/main/java/org/neo4j/driver/v1/Entry.java]: I almost always import `Entry` but I am not sure if I am representative either. `Pair`, is fine by me.

File: driver/src/test/java/org/neo4j/driver/v1/TypePlayTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package org.neo4j.driver.v1;
+
+import org.junit.Test;
+
+public class TypePlayTest
+{
```
Comment (zhenlineo) [inline driver/src/test/java/org/neo4j/driver/v1/TypePlayTest.java]: ? What's the purpose of this test?

File: driver/src/main/java/org/neo4j/driver/internal/util/Format.java [hunk 3]
```diff
@@ -33,9 +31,9 @@ private Format()
         throw new UnsupportedOperationException();
     }
 
-    public static <V> String formatProperties( Function<V, String> printValue,
-                                               int propertyCount,
-                                               Iterable<Property<V>> properties )
+    public static <V> String formatEntries( Function<V, String> printValue,
```
Comment (zhenlineo) [inline driver/src/main/java/org/neo4j/driver/internal/util/Format.java]: Do we still keep the idea of an `entry`?
Shall it be called `formatPairs`

File: driver/src/main/java/org/neo4j/driver/internal/util/Format.java [hunk 4]
```diff
@@ -33,9 +31,9 @@ private Format()
...
+                                            int propertyCount,
+                                            Iterable<Pair<String, V>> Entries )
```
Comment (zhenlineo) [inline driver/src/main/java/org/neo4j/driver/internal/util/Format.java]: `Entries` -> `entries` or `pairs`?

File: driver/src/main/java/org/neo4j/driver/internal/util/Format.java (hunk 3 above)
Comment (boggle) [inline driver/src/main/java/org/neo4j/driver/internal/util/Format.java]: Yep, that's an omission. Good catch.

**Comment by pontusmelke** [comment]
licenses...

**Pull request merged**